### ISSUE TITLE
Error: Multiple matches for name / ServiceAccount

### ISSUE THREAD

**Issue by looztra** [kubernetes-sigs/kustomize#1352]
## Structure

- `base` declares a `ServiceAccount`, no namespace is specified
- `overlays/integration` declares a `ClusterRoleBinding` refering the `ServiceAccount` in the `apps-integration` namespace
- `overlays/demo` declares a `ClusterRoleBinding` refering the `ServiceAccount` in the `apps-demo` namespace
- `overlays/both` is a composition of the two previous overlays

Git repository with test files : <https://github.com/looztra/kustomize-multibase-multiple-match-error>

## What works

- `kustomize build overlays/integration`
- `kustomize build overlays/demo`

## What fails but is expected to work

- `kustomize build overlays/both`

Error message :

```bash
╰─» asdf global kustomize v2.0.3
╰─» kustomize build overlays/both/
Error: Multiple matches for name ~G_v1_ServiceAccount|apps-integration|~P|faros|~S:
  [~G_v1_ServiceAccount|apps-integration|~P|faros|~S ~G_v1_ServiceAccount|apps-demo|~P|faros|~S]

╰─» asdf global kustomize v2.1.0
╰─» kustomize build overlays/both/
Error: string case - multiple matches for ~G_v1_ServiceAccount|~X|faros:
  [~G_v1_ServiceAccount|apps-demo|faros
 ~G_v1_ServiceAccount|apps-integration|faros
]

╰─» asdf global kustomize v3.0.2
╰─» kustomize build overlays/both/
Error: string case - multiple matches for ~G_v1_ServiceAccount|~X|faros:
  [~G_v1_ServiceAccount|apps-demo|faros
 ~G_v1_ServiceAccount|apps-integration|faros
]

```

## Context

Looked like #557 that has been fixed by #1344 but still fails with `v3.0.2` (no namePrefix/Suffix in my usecase)

Any idea?

**Comment by jbrette** [comment]
This seems to be fixed by this [PR](https://github.com/kubernetes-sigs/kustomize/pull/1325)

```bash
kustomize build overlays/both/
```

```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: faros
  namespace: apps-demo
---
apiVersion: v1
kind: ServiceAccount
metadata:
  name: faros
  namespace: apps-integration
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: faros-clustergittrackobjects-viewer--apps-demo
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: faros-clustergittrackobjects-viewer
subjects:
- kind: ServiceAccount
  name: faros
  namespace: apps-demo
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: faros-clustergittrackobjects-viewer--apps-integration
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: faros-clustergittrackobjects-viewer
subjects:
- kind: ServiceAccount
  name: faros
  namespace: apps-integration
```

**Comment by looztra** [comment]
I will give it a try asap, thanks!

**Comment by looztra** [comment]
I checked and it works, great job, thanks!

**Comment by looztra** [comment]
fixed in v3.0.3

thank you!
